### PR TITLE
fix(list):added test to list file on a new allocation

### PR DESCRIPTION
**Pull request by cnlangzi**
## Changes
- added test to list file on a new allocation

## PRs
- https://github.com/0chain/blobber/pull/605

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/getobjecttree  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/getobjecttree  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: tests/cli_tests/zboxcli_list_file_system_test.go [hunk 1]
```diff
@@ -503,6 +503,38 @@ func TestListFileSystem(t *testing.T) {
...
+		allocationID := setupAllocation(t, configPath)
+
+		output, err := listAllFilesInAllocation(t, configPath, createParams(map[string]interface{}{
+			"allocation": allocationID,
+		}), true)
+		require.Nil(t, err, "list files failed", strings.Join(output, "\n"))
+		require.Len(t, output, 1)
+
+		var listResults []climodel.ListFileResult
+		err = json.NewDecoder(strings.NewReader(output[0])).Decode(&listResults)
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/zboxcli_list_file_system_test.go:519]: Can this be replaced with Json.Unmarshal for consistency?

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]

File: tests/cli_tests/zboxcli_list_file_system_test.go [hunk 2]
```diff
@@ -503,6 +503,38 @@ func TestListFileSystem(t *testing.T) {
...
+		require.Nil(t, err, "Decoding list results failed\n", strings.Join(output, "\n"))
+
+		totalFiles := 0
+		totalFolders := 0
+		expectedTotalEntries := totalFolders + totalFiles
```
Comment (peterlimg) [inline tests/cli_tests/zboxcli_list_file_system_test.go:524]: The totalFolders and totalFiles are all 0, basically we only need to assert the returned `listResults` is 0.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: tests/cli_tests/zboxcli_list_file_system_test.go (hunk 2 above)
Comment (Kishan-Dhakan) [reply tests/cli_tests/zboxcli_list_file_system_test.go:524]: Agreed, @cnlangzi you can use `require.Empty()`

File: tests/cli_tests/zboxcli_list_file_system_test.go (hunk 1 above)
Comment (cnlangzi) [reply tests/cli_tests/zboxcli_list_file_system_test.go]: fixed

File: tests/cli_tests/zboxcli_list_file_system_test.go (hunk 2 above)
Comment (cnlangzi) [reply tests/cli_tests/zboxcli_list_file_system_test.go]: fixed

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**